Slim/Networking/Async/HTTP.pm

```diff
@@ -58,13 +59,21 @@ my $cookieJar;
 my $log = logger('network.asynchttp');
 
 __PACKAGE__->mk_accessor( rw => qw(
-	uri request response saveAs fh timeout maxRedirect
+	uri request response saveAs fh timeout maxRedirect socksAddr socksPort
 ) );
 
 sub init {
 	$cookieJar = HTTP::Cookies->new( file => catdir($prefs->get('cachedir'), 'cookies.dat'), autosave => 1 );
 }
 
+sub new {
+	my ($class, $args) = @_;
+	my $self = $class->SUPER::new;
+	$self->socksAddr( $args->{socksAddr} ); 
```

Comment:
Would it make sense to add some input value validation? Throw an error if `socksAddr` wasn't an IP or at least reasonable string?

Reply:
I'll look into that, I'm not very good at these this but I'll find some examples